dashboard/lib/services/user/password_resetter.rb

```diff
@@ -0,0 +1,49 @@
+module Services
+  module User
+    class PasswordResetter < Services::Base
+      attr_reader :email
+
+      def initialize(email:)
+        @email = email
+      end
+
+      def call
+        return user if user.errors.present?
+
+        if !user.nil? && user.authentication_options.any?(&:email?)
```

Comment:
```suggestion
        return unless user
        return user if user.errors.present?

        if user.authentication_options.any?(&:email?)
```

Reply:
This service object can't return nil, which is a mistake I was making previously. I committed changes to address this